### PR TITLE
Redesign kudos header Factual/front#12706

### PR DESCRIPTION
**Pull request by trisarahsorus**
## This PR...
- Adds the redesigned header to Kudos as a React component
- Closes Factual/front#12706

## Media
screen shot 2018-05-23 at 09 39 21

## Reviewers @joecox @nnsookwon 
My linter fixer went crazy on the files, sorry for the comma/whitespace changes everywhere T_T

### CODE REVIEW

File: app/assets/stylesheets/kudos/components/_header.scss [hunk 1]
```diff
@@ -0,0 +1,128 @@
+$stripe-width: 60px;
+$font-primary: Gotham SSm A, Gotham SSm B, Helvetica Neue, Helvetica, Arial, sans-serif;
```
Comment (nnsookwon) [inline app/assets/stylesheets/kudos/components/_header.scss]: I added a `$font-gotham` to variable.scss as you suggested in the other PR, if you want to use that instead

File: app/assets/stylesheets/kudos/components/_header.scss [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+.header {
+  display: block;
+  position: relative;
+}
+
+.header-stripes {
+  display: grid;
+  grid: repeat(3, $stripe-width) / repeat(5, 1fr);
```
Comment (joecox) [inline app/assets/stylesheets/kudos/components/_header.scss]: Sarah T living in 3018 while we all in 2018...

File: client/app/bundles/KudosApp/actions/actionCreators.jsx [hunk 3]
```diff
@@ -81,67 +81,63 @@ const createKudo = (receiverEmails, messageBody, onSuccess = null, onFailure = n
...
   return dispatch => {
-    dispatch(updatedKudo( id, message ))
-    dispatch(resetErrorMessage());
+    dispatch(updatedKudo(id, message))
+    dispatch(resetErrorMessage())
 
     return request({
       method: 'PATCH',
-      url: '/kudos/'+id,
+      url: '/kudos/' + id,
```
Comment (joecox) [inline client/app/bundles/KudosApp/actions/actionCreators.jsx]: Can we use backtick syntax?

File: client/app/bundles/KudosApp/components/Kudo.jsx [hunk 4]
```diff
@@ -96,67 +90,51 @@ export default class Kudo extends React.Component {
...
-        <div>
-          { this.state.editing ? <Save /> : <Edit /> }
-          {/* <Delete /> */}
-        </div>
-      ) : null
-    )
+    return this.postedByActiveUser() ? (
+      <div>
+        {this.state.editing ? <Save /> : <Edit />}
+        {/* <Delete /> */}
```
Comment (joecox) [inline client/app/bundles/KudosApp/components/Kudo.jsx]: I don't think we're planning on supporting delete, so let's just remove this and the TODO.

File: client/app/bundles/KudosApp/components/Kudo.jsx [hunk 5]
```diff
@@ -96,67 +90,51 @@ export default class Kudo extends React.Component {
...
   render() {
     return (
       <div className="kudo">
         <div className="content">
           <div className="sender">
-            <UserAvatar user={ this.props.kudo.giver } />
+            <UserAvatar user={this.props.kudo.giver} />
           </div>
-          <div className={ "receiver " + this.props.colorClass }>
+          <div className={'receiver ' + this.props.colorClass}>
```
Comment (joecox) [inline client/app/bundles/KudosApp/components/Kudo.jsx]: Can we use backtick syntax?

File: client/app/bundles/KudosApp/containers/KudosApp.jsx [hunk 6]
```diff
@@ -1,41 +1,43 @@
...
+import * as actionCreators from '../actions/actionCreators'
 
 // Simple example of a React "smart" component
-const KudosApp = ({ kudos, error, createKudo, isFetchingKudos, totalKudos }) => {
+const KudosApp = ({ user, kudos, error, createKudo, isFetchingKudos, totalKudos }) => {
   return (
     <div>
-      <ErrorBanner error={ error } />
-      <GiveKudo createKudo={ createKudo } />
+      <Header user={user} />
```
[the remark this answers is not in the document]
Comment (trisarahsorus) [reply client/app/bundles/KudosApp/containers/KudosApp.jsx]: Oop, good catch! :D

**Pull request merged**